src/auth_middleware.rs

```diff
@@ -100,7 +111,34 @@ impl<H: Handler> Handler for AuthHandler<H> {
     }
 }
 
+/// Handle JWT authentication on specified endpoints.
+///
+/// # Examples
+///
+/// Before passing it as middleware, you should specify the authenticated
+/// endpoints:
+///
+/// ```
+/// extern crate iron;
+/// extern crate foxbox_users;
+///
+/// fn main() {
+///     use foxbox_users::users_router::UsersRouter;
+///     use foxbox_users::auth_middleware::AuthMiddleware;
+///     use iron::prelude::{Chain, Iron};
+///
+///     let router = UsersRouter::init();
+///     let mut chain = Chain::new(router);
+///     chain.around(AuthMiddleware{
+///         auth_endpoints: vec![]
+///     });
+/// # if false {
```

Comment:
The embedded code is actually compiled and executed while `cargo test`. Preceding a line with `#` you can hide code. I want the next line `Iron::new(...)` to be compiled but never executed (otherwise `cargo test` will never finish as you're starting a server) so I enclose it in a never entering block. Nasty but working.